src/node_file.cc

```diff
@@ -3350,6 +3351,77 @@ static void CpSyncCheckPaths(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void CpSyncOverrideFile(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  CHECK_EQ(args.Length(), 4);  // src, dest, mode, preserveTimestamps
+
+  BufferValue src(isolate, args[0]);
+  CHECK_NOT_NULL(*src);
+  ToNamespacedPath(env, &src);
+
+  BufferValue dest(isolate, args[1]);
+  CHECK_NOT_NULL(*dest);
+  ToNamespacedPath(env, &dest);
+
+  int mode;
+  if (!GetValidFileMode(env, args[2], UV_FS_COPYFILE).To(&mode)) {
+    return;
+  }
+
+  bool preserve_timestamps = args[3]->IsTrue();
+
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kFileSystemRead, src.ToStringView());
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kFileSystemWrite, dest.ToStringView());
+
+  std::error_code error;
+
+  std::filesystem::remove(*dest, error);
+  if (error) {
+    std::string message = "operation not permitted, unlink";
+    return env->ThrowErrnoException(
+        EPERM, "unlink", message.c_str(), dest.out());
+  }
+
+  if (mode == 0) {
+    // if no mode is specified use the faster std::filesystem API
+    std::filesystem::copy_file(
+        *src, *dest, std::filesystem::copy_options::none, error);
+    if (error) {
+      return env->ThrowError(error.message().c_str());
```

Comment:
This doesn't have error code property as well. Can you throw appropriate error?

Reply:
I don't actually think that this can throw at this point since:
 - write permissions are guaranteed be the logic that comes before
 - the file path's validity is guaranteed because this function is only called when there is already a file there
 - the file is guaranteed not to exist because of the `remove` logic above
 
 This is more of a catch-all sort of line, just to make sure that if something does go wrong it is properly surfaced
 
So I wouldn't really know what error we could throw here... :confused:  any suggestion? (and/or is my reasoning above flawed?)

Reply:
Or, if 3382 executes but before 3391 file is executed a different process creates the same exact file, it will throw an error. I believe it's similar to "time of check to time of use" vulnerability.

Reply:
This should be addressed now (thanks a lot for the help @anonrig :heart_hands:)